crates/pallets/starknet/src/lib.rs

```diff
@@ -952,19 +957,19 @@ impl<T: Config> Pallet<T> {
                 parent_block_hash,
                 block_number,
                 global_state_root,
-                Felt252Wrapper::try_from(&sequencer_address).unwrap(),
+                sequencer_address,
                 block_timestamp,
                 transaction_count,
-                transaction_commitment.try_into().unwrap(),
+                Felt252Wrapper::from(transaction_commitment),
                 events.len() as u128,
-                event_commitment.try_into().unwrap(),
+                Felt252Wrapper::from(event_commitment),
                 protocol_version,
                 extra_data,
             ),
             // Safe because `transactions` is build from the `pending` bounded vec,
             // which has the same size limit of `MaxTransactions`
-            BoundedVec::try_from(transactions).unwrap(),
-            BoundedVec::try_from(receipts).unwrap(),
+            BoundedVec::try_from(transactions).expect("max(len(transactions)) <= MaxTransactions"),
+            BoundedVec::try_from(receipts).expect("max(len(receipts)) <= MaxTransactions"),
```

Comment:
`.expect` can still panic which is what we want to avoid in the runtime at all cost. Please look into `.map_err` instead

Reply:
> expect can still panic

agreed but the invariant here is that `len(receipts)` will always be <= `MaxTransaction` because of the way we are creating it.

 so this would never panic and thus is okay to `unwrap` it. 

i changed it to `except` just to make it more descriptive.

Reply:
Alright